pkgs/tools/misc/fluent-bit/default.nix

```diff
@@ -0,0 +1,23 @@
+{ pkgs, stdenv, fetchgit, cmake, ... }:
+
+stdenv.mkDerivation rec {
+  src = fetchgit {
+    url = https://github.com/fluent/fluent-bit;
+    sha256 = "0rmdbrhhrim80d0hwbz56d5f8rypm6h62ks3xnr0b4w987w10653";
+    rev = "8cc3a1887c3fcd6dd95a4a475fb213a0e399c222";
+  };
+  name = "fluentbit";
+  version = "1.0.6";
+  buildInputs = [ cmake ];
+  builder = ./builder.sh;
```

Comment:
I'm not seeing a specific reason you need your own builder so you can remove this and the default will take care of it for you.

Reply:
The package does not trivially build with the default builder. However, there are two problems with this builder.

1. `$out/bin` is filled with programs that do not belong in the package (`xxd-c`, `hello_world`, `onigmo-config`, etc.) while `$out/{include,lib}` is missing.

2. Builders do not compose and generally should be used as a last resort.

In this case, upstream assumed FHS location of systemd unit directory, which is read-only on NixOS, which causes an error during `installPhase`. Fixing that in `postPatch` phase will suffice:

```
  postPatch = ''
    substituteInPlace src/CMakeLists.txt \
      --replace /lib/systemd $out/lib/systemd
  '';
```